app/_how-tos/operator-controlplane-feature-gates-controllers.md

```diff
@@ -0,0 +1,232 @@
+  a: Use the `spec.featureGates` and `spec.controllers` fields in the ControlPlane resource.
+
+prereqs:
+  products:
+    - operator:
+        prereq_type: install
+---
+
+This guide explains how to configure feature gates and controllers for a ControlPlane in {{ site.ko_product_name }}. Feature gates allow you to enable or disable specific features, while controllers allow you to enable or disable specific resource reconciliation.
+
+## Feature Gates
+
+Feature gates control the availability of features in the ControlPlane. They follow the same concept as [Kubernetes feature gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/).
+
+### Configure Feature Gates
+
+To configure feature gates, use the `spec.featureGates` field in your ControlPlane resource:
+
+```yaml
+apiVersion: gateway-operator.konghq.com/v2beta1
+kind: ControlPlane
+metadata:
+  name: my-controlplane
+spec:
+  featureGates:
+  - name: GatewayAlpha
+    state: enabled
+  - name: FillIDs
+    state: enabled
+  dataplane:
+    type: managedByOwner
+```
+
+### Available Feature Gates
+
+The following feature gates are available:
+
+| Feature Gate | Default | Description |
+|--------------|---------|-------------|
+| `GatewayAlpha` | `false` | Enables alpha maturity Gateway API features |
+| `FillIDs` | `true` | Makes KIC fill in ID fields of Kong entities (Services, Routes, Consumers) to ensure stable IDs across restarts |
+| `RewriteURIs` | `false` | Enables the `konghq.com/rewrite` annotation |
+| `KongServiceFacade` | `false` | Enables KongServiceFacade CR reconciliation |
+| `SanitizeKonnectConfigDumps` | `true` | Enables sanitization of Konnect config dumps |
+| `FallbackConfiguration` | `false` | Enables generating fallback configuration when Kong Admin API returns entity errors |
+| `KongCustomEntity` | `true` | Enables KongCustomEntity CR reconciliation for custom Kong entities |
+
+{:.info}
+> **Note:** The `KongCustomEntity` feature gate requires `FillIDs` to be enabled, as custom entities require stable IDs for their foreign field references.
+
+## Controllers
+
+Controllers determine which Kubernetes resources the ControlPlane will reconcile. You can selectively enable or disable controllers based on your needs.
+
+### Configure Controllers
+
+To configure controllers, use the `spec.controllers` field in your ControlPlane resource:
+
+```yaml
+apiVersion: gateway-operator.konghq.com/v2beta1
+kind: ControlPlane
+metadata:
+  name: my-controlplane
+spec:
+  controllers:
+  - name: INGRESS_NETWORKINGV1
+    state: enabled
+  - name: SERVICE
+    state: enabled
+  - name: KONG_PLUGIN
+    state: enabled
+  - name: GWAPI_GATEWAY
+    state: disabled
+  - name: GWAPI_HTTPROUTE
+    state: disabled
+  dataplane:
+    type: managedByOwner
+```
+
+### Available Controllers
+
+The following controllers are available:
+
+#### Ingress Controllers
+| Controller Name | Description |
```

Comment:
Could we add the column to introduce whether the controller is enabled by default?

Reply:
Sure. Added https://github.com/Kong/developer.konghq.com/pull/2630/commits/f02a0296373ad402e00b5ae557651a5a44fcc2d9